twilight-model/src/guild/widget/mod.rs

```diff
@@ -0,0 +1,196 @@
+mod settings;
+
+pub use self::settings::GuildWidgetSettings;
+
+use crate::{
+    gateway::presence::Status,
+    id::{
+        marker::{ChannelMarker, GuildMarker, UserMarker},
+        Id,
+    },
+    user::{discriminator, DiscriminatorDisplay},
+    util::ImageHash,
+};
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, Deserialize, Eq, Hash, PartialEq, Serialize)]
+pub struct GuildWidget {
+    pub channels: Vec<GuildWidgetChannel>,
+    pub id: Id<GuildMarker>,
+    pub instant_invite: Option<String>,
+    pub members: Vec<GuildWidgetMember>,
+    pub name: String,
+    pub presence_count: u64,
+}
+
+#[derive(Clone, Debug, Deserialize, Eq, Hash, PartialEq, Serialize)]
+pub struct GuildWidgetChannel {
+    pub id: Id<ChannelMarker>,
+    pub name: String,
+    pub position: i64,
+}
+
+#[derive(Clone, Debug, Deserialize, Eq, Hash, PartialEq, Serialize)]
+pub struct GuildWidgetMember {
+    pub avatar: Option<ImageHash>,
+    pub avatar_url: Option<String>,
+    #[serde(with = "discriminator")]
+    pub discriminator: u16,
+    pub id: Id<UserMarker>,
```

Comment:
This ID may be zero when it is anonymized, which will cause a deserialsation failure akind to the one in https://github.com/twilight-rs/twilight/issues/1979.

I think either we should have a special nullable id or we should just see this as a string.